### PR TITLE
Star6 01

### PR DESCRIPTION
**Pull request by perevbnlgov**
Outdated StKF* and KFP* and KFV* removed. Code which is used it also removed.
According LinDef is changed. By request of Y.Fisyak

### CODE REVIEW

**Review by ullrich-bnl — APPROVED** [review]

**Comment by starsdong** [comment]
Victor, thank you very much for pushing this forward. One quick question, the changes proposed in StBFChain in this PR don't seem to be related to the need for ROOT6. Or did I miss something here?
On the proposed changes to mgr/, Dmitri or Grigory would be nice to hear your comments. Thanks

File: StRoot/StBFChain/doc/BFC.h [hunk 1]
```diff
@@ -0,0 +1,1693 @@
+Bfc_st BFC[] = {
```
Comment (klendathu2k) [inline StRoot/StBFChain/doc/BFC.h:1]: Files under StBFChain/doc are generated during the build process.  They should not be committed into this directory.  (Use "git rm" to remove the files).

File: StRoot/StBFChain/doc/BFC2.h [hunk 2]
```diff
@@ -0,0 +1,1693 @@
+Bfc_st BFC[] = {
```
Comment (klendathu2k) [inline StRoot/StBFChain/doc/BFC2.h:1]: Remove

File: StRoot/StBFChain/doc/StBFChain.cxx_doc [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// @(#)StRoot/StBFChain:$Name:  $:$Id: StBFChain.cxx_doc,v 1.180 2021/03/30 14:45:42 genevb Exp $
+// @(#)StRoot/StBFChain:$Name:  $:$Id: StBFChain.cxx,v 1.670 2021/03/21 03:32:28 genevb Exp $
```
Comment (klendathu2k) [inline StRoot/StBFChain/doc/StBFChain.cxx_doc:1]: ... looks like we are replacing a newer revision with an older one (1.670 vs 1.180)... but again, these files are generated at compile time.  Should remove this.

File: mgr/ConsDefs.pm [hunk 4]
```diff
@@ -173,8 +174,9 @@
     $CXXFLAGS = $CFLAGS = $LDFLAGS = $SOFLAGS = "$XMACHOPT";
 
     $CXXFLAGS      .= " -fPIC -w";
+$CXXFLAGS .= "-fdiagnostics-color=always -msse4.2 -std=c++11 -pipe  -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -pthread -DNDEBUG";
```
Comment (klendathu2k) [inline mgr/ConsDefs.pm:177]: Not clear why we need the -msse4.2 option.  Is this for root 6, or are we just picking up a change for the heck of it?

Also... defines NDEBUG... which I am not convinced is the right thing to do here.  (Doesn't this break logic for creating debug vs nodebug builds?  Has this change to the CXXFLAGS been tested... beyond "it compiles, it runs, and plots can be made that sorta kinda look right")

File: mgr/RootCint.pl [hunk 5]
```diff
@@ -437,9 +440,10 @@
     #	$cmd  = "rootcint -f $Cint_cxx -c  -D__NO_STRANGE_MUDST__ -DROOT_CINT -D__ROOT__ $CPPFLAGS $h_files";
     #	print "cmd (+extra) = ",$cmd,"\n";
     #} else {
-    	$cmd  = "rootcint -f $Cint_cxx -c -DROOT_CINT -D__ROOT__ $CPPFLAGS $h_files";
+##    	$cmd  = "rootcint  -f $Cint_cxx -c -DROOT_CINT -D__ROOT__ $CPPFLAGS $h_files";
+    	$cmd  = "rootcling -rmf $Cint_map -f $Cint_cxx  -DROOT_CINT -D__ROOT__ $CPPFLAGS $h_files";
```
Comment (klendathu2k) [inline mgr/RootCint.pl:444]: Does this work for our ROOT5 builds as well?  (This commit _cannot_ break the ROOT5 build.)

File: mgr/RootCint.pl [hunk 6]
```diff
@@ -437,9 +440,10 @@
...
      	print "cmd (normal)= ",$cmd,"\n";
-    #	die;
+##    	die;
```
Comment (klendathu2k) [inline mgr/RootCint.pl:446]: Yes.  Let's _really_ make sure this line is commented out ;)

File: mgr/RootCint.pl (hunk 5 above)
Comment (klendathu2k) [reply mgr/RootCint.pl:444]: ... probably should have logic testing against the root version we compile against, and use root5/rootcint versus root6/rootcling.

File: StRoot/StBFChain/StBFChain.cxx [hunk 7]
```diff
@@ -638,11 +638,14 @@ Int_t StBFChain::Instantiate()
     }
 
     if (maker == "StStrangeMuDstMaker" && GetOption("CMuDST")&& GetOption("StrngMuDST") ) {
-      TString cmd(Form("StStrangeMuDstMaker *pSMMk = (StStrangeMuDstMaker*) %p;",mk));
-      cmd += "pSMMk->DoV0();";                                 // Set StrangeMuDstMaker parameters
-      cmd += "pSMMk->DoXi();";
-      cmd += "pSMMk->DoKink();";
-      cmd += "pSMMk->SetNoKeep();";                            // Set flag for output OFF
+printf ("UUU Mk = %p\n",mk);
```
Comment (klendathu2k) [inline StRoot/StBFChain/StBFChain.cxx:641]: This introduces a change to the StStrangeMuDstMaker configuration... and also creates meaningless output.

The old version needs to be restored.

**Review by klendathu2k — CHANGES_REQUESTED** [review]
Thank you for pruning this down to a minimal set of changes.  There are several issues identified, and a couple of questions.  

But I note that the CI builds have failed for both ROOT5 and ROOT6.  Regardless of getting two approvals, the code must actually compile under both ROOT5 and ROOT6... (under both cons and the CI builds).

**Comment by plexoos** [comment]
> On the proposed changes to mgr/, Dmitri or Grigory would be nice to hear your comments. Thanks

I agree with the Jason's remarks. Before we continue with any further discussion the code should compile.

File: StRoot/StBFChain/doc/BFC.h (hunk 1 above)
Comment (perevbnlgov) [reply StRoot/StBFChain/doc/BFC.h:1]: Yes, I know, I am trying not to touch these doc files, ut they are arised again. I did not find how to ignore them.
Best of all to remove them at all. Automatically created files should not be in ane CVS or Git repository.
They shoukd be created in some temporary directories like .o, .so ...

**Comment by perevbnlgov** [comment]
pull request Star6_01 closed without merging. Wrong request.

**Pull request closed without merging**